hepcrawl/spiders/desy_spider.py

```diff
@@ -277,24 +274,26 @@ def _get_marcxml_records(response_body):
 
         return [etree.tostring(item) for item in list_items]
 
-    def _hep_records_from_marcxml(self, marcxml_records):
-        def _create_json_record(xml_record):
+    def _parsed_item_from_marcxml(self, marcxml_records):
+        def _get_crawl_result(xml_record):
             app = Flask('hepcrawl')
             app.config.update(
                 self.settings.getdict('MARC_TO_HEP_SETTINGS', {})
             )
             with app.app_context():
+                item = ParsedItem(record={}, record_format='hep')
                 try:
-                    hep_record = marcxml2record(xml_record)
+                    item.record = marcxml2record(xml_record)
                 except Exception as e:
-                    return {'xml_record': xml_record, 'error': repr(e),
-                            'traceback': traceback.format_tb(sys.exc_info()[2])}
-                
-            return hep_record
+                    item.exception = repr(e)
+                    item.traceback = traceback.format_tb(sys.exc_info()[2])
+                    item.source_data = xml_record
```

Comment:
instead of creating the object and then changing the properties, try to create at the same time (either with the init, or even creating a subclass `ErroredParsedItem`)